app/src/componentsV2/CodeEditor/components/EditorV2/components/VariablePopOver/hooks/useCreateVariable.ts

```diff
@@ -0,0 +1,127 @@
+import { useState, useCallback } from "react";
+import { VariableScope } from "backend/environment/types";
+import { CreateVariableFormData } from "../types";
+import { useCommand } from "features/apiClient/commands";
+import { useActiveEnvironment } from "features/apiClient/hooks/useActiveEnvironment.hook";
+import { runtimeVariablesStore } from "features/apiClient/store/runtimeVariables/runtimeVariables.store";
+import { toast } from "utils/Toast";
+import { useApiClientFeatureContext } from "features/apiClient/contexts/meta";
+
+interface UseCreateVariableResult {
+  createVariable: (data: CreateVariableFormData) => Promise<void>;
+  isCreating: boolean;
+  error: string | null;
+}
+
+export const useCreateVariable = (collectionId?: string): UseCreateVariableResult => {
+  const [isCreating, setIsCreating] = useState(false);
+  const [error, setError] = useState<string | null>(null);
+
+  const {
+    env: { patchEnvironmentVariables },
+    api: { patchCollectionVariables },
+  } = useCommand();
+
+  const activeEnvironment = useActiveEnvironment();
+  const { repositories } = useApiClientFeatureContext();
+  const { environmentVariablesRepository } = repositories;
+
+  const createVariable = useCallback(
+    async (data: CreateVariableFormData) => {
+      setIsCreating(true);
+      setError(null);
+
+      try {
+        const { variableName, scope, type, initialValue, currentValue } = data;
+
+        const variableData = {
+          [variableName]: {
+            id: 0, // Will be set by the store
+            type,
+            syncValue: initialValue ?? "",
+            localValue: currentValue ?? "",
+            isPersisted: true as const,
```

Comment:
Add this decision in linear that we are persisting all the variables, since by nature `runtime` variables are session only.